substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -1,94 +1,164 @@
+
+    pub const GRID_LOCK_ID: LockIdentifier = *b"gridlock";
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    // Version constant that referenced the struct version
+    pub const CONTRACT_VERSION: u32 = 3;
+
+    #[pallet::storage]
+    #[pallet::getter(fn contracts)]
+    pub type Contracts<T: Config> = StorageMap<_, Blake2_128Concat, u64, Contract, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn contract_billing_information_by_id)]
+    pub type ContractBillingInformationByID<T: Config> =
+        StorageMap<_, Blake2_128Concat, u64, ContractBillingInformation, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn node_contract_resources)]
+    pub type NodeContractResources<T: Config> =
+        StorageMap<_, Blake2_128Concat, u64, ContractResources, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn node_contract_by_hash)]
+    pub type ContractIDByNodeIDAndHash<T: Config> =
+        StorageDoubleMap<_, Blake2_128Concat, u32, Blake2_128Concat, Vec<u8>, u64, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn active_node_contracts)]
+    pub type ActiveNodeContracts<T: Config> =
+        StorageMap<_, Blake2_128Concat, u32, Vec<u64>, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn contract_to_bill_at_block)]
+    pub type ContractsToBillAt<T: Config> =
+        StorageMap<_, Blake2_128Concat, u64, Vec<u64>, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn contract_number_of_cylces_billed)]
+    pub type ContractLock<T: Config> =
+        StorageMap<_, Blake2_128Concat, u64, types::ContractLock<BalanceOf<T>>, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn contract_id_by_name_registration)]
+    pub type ContractIDByNameRegistration<T: Config> =
+        StorageMap<_, Blake2_128Concat, Vec<u8>, u64, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn active_rent_contracts)]
+    pub type ActiveRentContractForNode<T: Config> =
+        StorageMap<_, Blake2_128Concat, u32, Contract, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn contract_id)]
+    pub type ContractID<T> = StorageValue<_, u64, ValueQuery>;
+
+    #[pallet::config]
+    pub trait Config:
+        frame_system::Config
+        + pallet_timestamp::Config
+        + pallet_balances::Config
+        + pallet_tfgrid::Config
     {
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+        type Currency: LockableCurrency<Self::AccountId>;
+        type StakingPoolAccount: Get<Self::AccountId>;
+        type BillingFrequency: Get<u64>;
+        type DistributionFrequency: Get<u16>;
+        type GracePeriod: Get<u64>;
+        type WeightInfo: WeightInfo;
+        type NodeChanged: ChangeNode;
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
```

Comment:
While we're at it, let's start putting doc comments on event enum variants. This should be a small description of what it means if this event is encountered, as well as what the arguments are (as used by the "official" substrate crates, like balances). For example, NodeContractCancelled has 3 integer arguments, though it is not descriptive what this means without comment

Reply:
Also lets already indicate which variants are deprecated, so we can remove them for some cleanup in the future